src/qt/bitcoingui.cpp

```diff
@@ -437,9 +437,9 @@ void BitcoinGUI::createToolBars()
 
 void BitcoinGUI::setClientModel(ClientModel *_clientModel)
 {
-    this->clientModel = _clientModel;
     if(_clientModel)
     {
+        this->clientModel = _clientModel;
```

Comment:
Why this change? Should update `this->clientModel` event if `_clientModel` is null.

Reply:
Because now the main thread is not blocked, network events can trigger a GUI update. If the clientModel is set to 0 or null at the time, the application will crash.
Example: When number of connections change, BitcoinGUI::updateNetworkState() gets invoked and this method assumes that clientModel is valid.
With this change  BitcoinGUI clientModel is cleared only after the RPCExecutor thread joins and shutdown resumes in the main thread.

Reply:
> BitcoinGUI::updateNetworkState() gets invoked and this method assumes that clientModel is valid.

Then those need to be fixed IMO. With your change calling `setClientModel(nullptr)` on the `BitcoinGUI` instance is a noop.